src/main/resources/tailormap-api.yaml

```diff
@@ -816,3 +836,120 @@ paths:
             text/plain:
               schema:
                 type: string
+
+  /app/{appId}/layer/{layerId}/download/{format}:
+    summary: 'Use this endpoint to download sets of features in specified formats.'
+    post:
+      operationId: 'download'
+      description: 'retrieve features that fulfill the requested conditions (parameters).'
+      security:
+        - formAuth: [ ]
+      parameters:
+        - description: 'application id'
+          in: path
+          name: appId
+          required: true
+          schema:
+            type: integer
+            format: int64
+        - description: 'application layer id'
+          in: path
+          name: layerId
+          required: true
+          schema:
+            type: integer
+            format: int64
+        - description: 'download format, must equal one of the enum values'
+          in: path
+          name: format
+          required: true
+          schema:
+            type: string
+            enum:
+              - 'excel'
+              - 'geojson'
+              - 'geopackage'
+              - 'shapefile'
+              - 'kml'
+        - description: '
+            A filter to be applied, possibly in combination with any other request parameters. 
+            The filter is an ECQL string, see [ECQL reference](https://docs.geoserver.org/latest/en/user/filter/ecql_reference.html).
+            '
+          in: query
+          name: filter
+          required: false
+      responses:
+        '202':
+          description: 'OK'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DownloadStatusResponse'
+
```

Comment:
Optionally a cancel download request can also be added. The frontend should of course properly send this when the user isn't interested anymore (for example, when it's taking too long).